examples/01-filter/distance-between-surfaces.py

```diff
@@ -87,12 +87,45 @@ def hill(seed):
 from scipy.spatial import KDTree
 
 tree = KDTree(h1.points)
-d, idx = tree.query(h0.points )
-h0["distances"] = d
-np.mean(d)
+d_kdtree, idx = tree.query(h0.points )
+h0["distances"] = d_kdtree
+np.mean(d_kdtree)
 
 ###############################################################################
 p = pv.Plotter()
 p.add_mesh(h0, scalars="distances", smooth_shading=True)
 p.add_mesh(h1, color=True, opacity=0.75, smooth_shading=True)
 p.show()
+
+
+###############################################################################
+# Exact closest distance
+# ++++++++++++++++++++++
+#
+# The :func:`pyvista.DataSet.find_closest_cell` filter can be used to determine
+# the point on a surface that is closest to a set of points.  This method finds
+# points inside cells that are closest to the supplied points.  This provides
+# the exact closest distance from one mesh to the other. In this case, the
+# distances are calculated from points of ``h0`` to the entire mesh of ``h1``.
+
+_, closest_points = h1.find_closest_cell(h0.points, return_closest_point=True)
+
+d_exact = np.linalg.norm(h0.points - closest_points, axis=1)
+
+h0["distances"] = d_exact
+np.nanmean(d_exact)
```

Comment:
`d_exact` can't have NaNs, can it?

Reply:
I followed the pattern from above.  I will check if we can remove this from both places.

Reply:
Above I can only see `np.nanmean` for `h0["distances"]` which has nans due to initialisation. But the values we are assigning probably do not. I haven't checked yet.

Reply:
You are right, I didn't read close enough.